api/role/schema.py

```diff
@@ -18,25 +21,43 @@ class Arguments:
     role = graphene.Field(Role)
 
     def mutate(self, info, **kwargs):
-        role = RoleModel(**kwargs)
-        payload = {
-            'model': RoleModel, 'field': 'role', 'value':  kwargs['role']
-            }
-        with SaveContextManager(role,  'Role', payload):
-            return CreateRole(role=role)
+        try:
+            role = RoleModel(**kwargs)
+            payload = {
+                'model': RoleModel, 'field': 'role', 'value':  kwargs['role']
+                }
+            query = Role.get_query(info)
+            role_name = query.filter(
```

Comment:
kindly remove this code because its redundant with the presence of the context manager

Reply:
After the implementation of my feature, even after running the migration afresh, the query wasn't giving the intended error message even though it wasn't creating the roles with the same name. So the context manager was preventing creation of roles with duplicate names but it wasn't giving the error message(instead It was showing that the role was created, but in real sense it wasn't), so to give the error message I had to to do the check with this code explicitly. I hope that makes sense @RachaelNantale